docs/pages/kubernetes-access/register-clusters/iam-joining.mdx

```diff
@@ -0,0 +1,295 @@
+```
+
+At this point, the IAM role is ready to be used by the Teleport Kubernetes Agent Service Account.
+
+</TabItem>
+</Tabs>
+
+## Step 2/3. Create the AWS joining token
+
+Configure your Teleport Auth Server with a special dynamic token which will
+allow agents from your AWS account to join your Teleport cluster using the roles
+defined.
+
+Under the hood, Kubernetes Agents will prove that they are running in your AWS account by
+sending a signed Identity Document which matches an allow rule
+configured in your AWS joining token.
+
+Create the following `token.yaml` with an `allow` rule specifying your AWS
+account and the AWS ARN the agents will be running as.
+
+```code
+$ cat >token.yaml <<EOF
+kind: token
+version: v2
+metadata:
+  # the token name is not a secret because instances must prove that they are
+  # running in your AWS account to use this token
+  name: <Var name="kube-iam-token"/>
+  # set a long expiry time, the default for tokens is only 30 minutes
+  expires: "3000-01-01T00:00:00Z"
+spec:
+  # use the minimal set of roles required
+  roles: [Kube]
+  # set the join method allowed for this token
+  join_method: iam
+  allow:
+  # aws_arn is optional and allows you to restrict the IAM role of joining Agents
+  # to a specific IAM role
+  - aws_account: "$account_id"
+    aws_arn: "arn:aws:sts::$account_id:assumed-role/<Var name="kube-iam-role"/>/*"
+EOF
+```
+
+Run `tctl create token.yaml` to create the token.
+
+## Step 3/3. Deploy the Teleport Kubernetes Agent
+
+(!docs/pages/kubernetes-access/helm/includes/helm-repo-add.mdx!)
+
+Switch `kubectl` to the Kubernetes cluster and run:
+
+```code
+# Deploy a Kubernetes agent. It dials back to the Teleport cluster tele.example.com.
+$ CLUSTER=iam-cluster
+$ PROXY=tele.example.com:443
+$ helm install teleport-agent teleport/teleport-kube-agent \
+  --set kubeClusterName=${CLUSTER?} \
+  --set proxyAddr=${PROXY?} \
+  --set joinParams.method=iam \
+  --set joinParams.tokenName=<Var name="kube-iam-token"/> \
+  --set serviceAccount.create=false # do not create a new service account # \
+  --set serviceAccount.name=<Var name="teleport-kube-agent-sa"/> # use the existing service account # \
+  --create-namespace \
+  --namespace=<Var name="teleport-agent"/> \
+  --version (=teleport.version=)
+```
+
+List connected clusters using `tsh kube ls` and switch between
+them using `tsh kube login`:
+
+```code
+$ tsh kube ls
+
+# Kube Cluster Name Selected
+# ----------------- --------
+# iam-cluster
+
+# kubeconfig now points to the iam-cluster cluster
+$ tsh kube login iam-cluster
+# Logged into Kubernetes cluster "iam-cluster". Try 'kubectl version' to test the connection.
+
+# kubectl command executed on `iam-cluster` but is routed through the `tele.example.com` cluster.
+$ kubectl get pods
+```
+
+<Details title="Not seeing Kubernetes clusters?">
+
```

Comment:
I think this could use 1-2 sentences before the partial to introduce it, i.e., if you're not seeing clusters, it's probably because your role doesn't include the correct permissions

Reply:
Done in 4e408ff08eccc61fbbb073110df7e05c96e5ca5c